docker-bake.hcl

```diff
@@ -579,9 +591,18 @@ function "pkgPlatforms" {
     docker-engine = ["linux/amd64", "linux/arm/v6", "linux/arm/v7", "linux/arm64", "linux/ppc64le", "linux/s390x", "windows/amd64", "windows/arm64"]
     # https://github.com/docker/model-cli/blob/301126afc8ef4b8330de56db5d2889ddbc978022/Makefile#L36-L40
     model = ["darwin/amd64", "darwin/arm64", "linux/amd64", "linux/arm64", "linux/arm/v7", "windows/amd64", "windows/arm64"]
+    # https://github.com/docker/cagent/blob/0538a5915160991e39c38d5f446d0507b59a8369/Taskfile.yml#L65
+    # + https://github.com/docker/cagent/issues/85
+    cagent = ["darwin/amd64", "darwin/arm64", "linux/amd64", "linux/arm64", "windows/amd64", "windows/arm64"]
   }, pkg, [])
 }
 
+# Returns the list of supported platforms for static packages for a given package.
+function "staticPkgPlatforms" {
+  params = [pkg]
+  result = pkg == "cagent" ? [] : pkgPlatforms(pkg)
```

Comment:
Looking at the Dockerfile: https://github.com/docker/cagent/blob/ab6f7b276b29434623daf733a3d0b524d2d6601d/Dockerfile

I think we can make "static" builds, similar to what has be done for credential helpers: https://github.com/docker/packaging/blob/29ca532a5aba620ea87b3b62e57f465073b617f2/pkg/credential-helpers/Dockerfile#L158-L210

Reply:
Windows static binaries need zig and I haven't figured it out why yet 😅 

Also we wouldn't be deploying any static binaries to download.docker.com yet.

My main intent here is to provide `cagent` packages for Linux distros, so the static binaries are not really needed.

The static binaries are already published in releases https://github.com/docker/cagent/releases, so let's not duplicate that.

Reply:
https://github.com/docker/cagent/pull/192 should help so we don't need CGO anymore and can do proper static builds